### PR TITLE
[2.1.20] forced wait on some async multichannel tasks

### PR DESCRIPTION
**Pull request by mzieg**
Added awaits on async tasks what were running in parallel but ought'nt of.

### CODE REVIEW

File: WasatchNET/EEPROM.cs [hunk 1]
```diff
@@ -1951,7 +1951,7 @@ void enforceReasonableDefaults()
                 wavecalCoeffs[4] = 0;
             }
 
-            if (minIntegrationTimeMS < 1)
+            if (minIntegrationTimeMS < 1 || minIntegrationTimeMS == 65535)
```
Comment (mzieg) [inline WasatchNET/EEPROM.cs:1954]: Honestly, I should probably make that >5000.  No WP spectrometer is going to have a minimum integration more than 5sec (or even 1sec).

File: WasatchNET/MultiChannelWrapper.cs [hunk 2]
```diff
@@ -340,10 +342,16 @@ public async void startAcquisition()
         /// total acquisition time is close to minimal.
         /// </remarks>
         /// <param name="sendTrigger">whether to automatically raise the HW trigger (default true)</param>
-        public List<ChannelSpectrum> getSpectra(bool sendTrigger=true)
+        public async Task<List<ChannelSpectrum>> getSpectra(bool sendTrigger=true)
         {
             if (sendTrigger)
-                startAcquisition();
+            {
+                if (!await startAcquisition())
```
Comment (mzieg) [inline WasatchNET/MultiChannelWrapper.cs:349]: This was the fundamental bug which Cullon uncovered in his MultiChannel testing.  I wasn't actually waiting on the trigger pulse to complete (rise, level, then fall) before moving on to reading the spectra whose integrations were presumably started by the pulse.

File: WasatchNET/Spectrometer.cs [hunk 3]
```diff
@@ -2119,28 +2119,28 @@ uint[] readImage(UsbEndpointReader spectralReader, int pixelsPerEndpoint)
...
 
                 bytesReadThisEndpoint += bytesRead;
-                logger.debug("readSubspectrum: bytesReadThisEndpoint now {0}", bytesReadThisEndpoint);
+                logger.debug("readImage: bytesReadThisEndpoint now {0}", bytesReadThisEndpoint);
 
                 if (bytesReadThisEndpoint == 0 && !triggerWasExternal)
                 {
-                    logger.error("readSpectrum: read nothing (timeout?)");
-                    return null; //  break; // should be return null
+                    logger.error("readImage: read nothing (timeout?)");
```
Comment (mzieg) [inline WasatchNET/Spectrometer.cs:2137]: These changes were necessitated when I found 3 separate functions that could generate "readSpectrum: read nothing (timeout?)"

File: WasatchNET/MultiChannelWrapper.cs (hunk 2 above)
Comment (trestoh) [reply WasatchNET/MultiChannelWrapper.cs:349]: I assume this goes along with the changes to Form1? I'll need to dig in once I start doing async await in WPSC

File: WasatchNET/Spectrometer.cs (hunk 3 above)
Comment (trestoh) [reply WasatchNET/Spectrometer.cs:2137]: If this is the area scan function...it needs an overhaul anyway once we get the firmware

**Pull request merged**